System.IO.Streams/MemoryStream.cs

```diff
@@ -464,11 +465,21 @@ private bool EnsureCapacity(int value)
                     newCapacity = CapacityDefaultSize;
                 }
 
+                if (value > MemStreamMaxLength)
+                {
+                    throw new ArgumentOutOfRangeException("value", $"MemoryStream can't be more than {MemStreamMaxLength} bytes");
```

Comment:
To save flash on deployment we tend to NOT have messages in the exceptions. Instead the explanation on the exception reason is clearly indicated in the Intellisense comment.
Can you please move this sentence there?

Reply:
Ahh, that's why that is.
Are there any plans to change this? Possibly by having a flag to set whether to include the messages in the firmware or something like that? On something like ESP32, there is 4 - 16MB of flash, a handful of exception messages aren't going to fill that.

I found that to be one of the somewhat irritating limitations of NF because sometimes I feel like explanations are just missing entirely and I have to look through the library code to see which case could have possibly caused the exception I'm getting

Reply:
It would require to first move to SDK style projects so that with a flag, 2 libraries can be build and based on some other elements the good one chosen transparently at build time. But it's not going to happen soon.
Keep in mind embedded development is a matter of choices and trade off. Those messages are useful for the debug phase only, reason why they've been addedd as description in the Intellisense to mitigate the stripping out from the libs.
Now as José pointed out, if a message is super important, we'll keep it but in general, we'll try not to.

Reply:
@patagonaa what you are suggesting is doable, one way is what Laurent explained above. Now, this would make distribution more complicated (and potentially not developer friendly) because one would find two version of the same library and it would be confusing. 
Still, I totally get your point and it has happened to me too. Sometimes one exception type can be originated in two or more different situations and then, one has to look into the code to try finding out why.
Fortunatly, in the vast majority of the situations, there is a one to one mapping, and then the Intellisense comments can point you in the right direction. 😉